middle_end/flambda2/simplify/simplify_expr.ml

```diff
@@ -69,6 +69,13 @@ let simplify_toplevel_common dacc simplify ~params ~return_continuation
         let uacc =
           UA.create ~flow_result ~compute_slot_offsets:true uenv dacc
         in
+        let uacc =
+          if not
+               (Named_rewrite_id.Map.is_empty
+                  flow_result.mutable_unboxing_result.let_rewrites)
```

Comment:
This could be beneficial for things that are not related to unboxing. Since this should be quite uncommon, it should be ok to do this in all the cases.